pkg/plugins/taskplugin/taskplugin.go

```diff
@@ -58,24 +59,29 @@ func (r PluginExecutor) ValidConfig(baseConfig json.RawMessage, config json.RawM
 }
 
 // Exec performs the action implemented by the executor
-func (r PluginExecutor) Exec(stepName string, baseConfig json.RawMessage, config json.RawMessage, ctx interface{}) (interface{}, interface{}, error) {
+func (r PluginExecutor) Exec(stepName string, baseConfig json.RawMessage, config json.RawMessage, ctx interface{}) (interface{}, interface{}, map[string]string, []string, error) {
 	var cfg interface{}
 
 	if r.configFactory != nil {
 		cfg = r.configFactory()
 		if len(baseConfig) > 0 {
 			err := utils.JSONnumberUnmarshal(bytes.NewReader(baseConfig), cfg)
 			if err != nil {
-				return nil, nil, errors.Annotate(err, "failed to unmarshal base configuration")
+				return nil, nil, nil, nil, errors.Annotate(err, "failed to unmarshal base configuration")
 			}
 		}
 		err := utils.JSONnumberUnmarshal(bytes.NewReader(config), cfg)
 		if err != nil {
-			return nil, nil, errors.Annotate(err, "failed to unmarshal configuration")
+			return nil, nil, nil, nil, errors.Annotate(err, "failed to unmarshal configuration")
 		}
 	}
-
-	return r.execfunc(stepName, cfg, ctx)
+	var newTags map[string]string
+	var tagsToRemove []string
+	if r.tagsFunc != nil {
+		newTags, tagsToRemove = r.tagsFunc(cfg, ctx)
```

Comment:
The tag func should probably be called after the main exec func, to be able to apply tags based on the result.
To that end, it should have access to the output/metadata/err values.

Reply:
I assumed that the config and context inputs were the only source for the tags mutation func, but yes.

Reply:
Something like that yes. Maybe err too ?

Reply:
Done.